### PR TITLE
cmd/trace-agent: unify command line flags

### PR DESCRIPTION
**Pull request by gbbr**
### Description

This change modifies the trace-agent CLI to use [`cobra`](https://github.com/spf13/cobra) in order to make it consistent with the core agent in an attempt to unify the command-line experience, enable POSIX flags and allow extracting better information for support.

```
trace-agent is Datadog's APM agent. It is comprised of an HTTP API which processes incoming
payloads representing tracing data. It augments them with stats and extracts Trace Search &
Analytics events, forwarding them to the Datadog API. The trace-agent is usually bundled and
run with the Datadog Agent package.

Usage:
  trace-agent [command]

Available Commands:
  config      Prints the config used by a running trace-agent
  help        Help about any command
  run         Run the agent
  status      Print the current status
  version     Print version information

Flags:
  -c, --cfgpath string   path to directory containing datadog.yaml (default "/opt/datadog-agent/etc/datadog.yaml")
  -h, --help             help for trace-agent

Use "trace-agent [command] --help" for more information about a command.
```

### Flags on Windows

Additional flags are made available on Windows:
```
Available Commands:
  ...
  install-service   Installs the trace-agent in the service control manager.
  start-service     Starts the trace-agent service
  stop-service      Stops the trace-agent service
  uninstall-service Uninstalls the trace-agent from the service control manager.
```

### TODO

- [x] Add copyright
- [x] Test stats output to file is still working with `benchmarking` build tag
- [ ] Test on Windows
- [x] Add `config` command to display runtime config
- [ ] Add `hostname` command to display hostname in use.

### CODE REVIEW

File: cmd/trace-agent/main_windows.go [hunk 1]
```diff
@@ -66,87 +76,32 @@ func (m *myservice) Execute(args []string, r <-chan svc.ChangeRequest, changes c
...
 		fmt.Printf("failed to determine if we are running in an interactive session: %v", err)
 	}
 	if !isIntSess {
-		runService(false)
-		return
-	}
-	// sigh.  Go doesn't have boolean xor operator.  The options are mutually exclusive,
-	// make sure more than one wasn't specified
-	optcount := 0
-	if flags.Win.InstallService {
```
Comment (palazzem) [inline cmd/trace-agent/main_windows.go]: Are we breaking anything on Windows, or all of this has been addressed differently?

Comment (gbbr) [reply cmd/trace-agent/main_windows.go]: They are different commands now. I'm hoping @derekwbrown will take a look to ensure all is well. I'll also be trying to install a local `.msi` on my VM.

File: pkg/trace/agent/run.go [hunk 2]
```diff
@@ -30,14 +24,13 @@ const messageAgentDisabled = `trace-agent not enabled. Set the environment varia
 DD_APM_ENABLED=true or add "apm_config.enabled: true" entry
 to your datadog.yaml. Exiting...`
 
+// ConfigPath specifies the path to the configuration file. It is set via the
+// command's -c / --cfgpath persistent flag.
```
Comment (palazzem) [inline pkg/trace/agent/run.go]: Is this a breaking change from our `-config` right?

File: pkg/trace/agent/run.go [hunk 3]
```diff
@@ -70,27 +56,6 @@ func Run(ctx context.Context) {
 
 	defer watchdog.LogOnPanic()
 
-	if flags.CPUProfile != "" {
```
Comment (palazzem) [inline pkg/trace/agent/run.go]: Are we offering CPU/Mem profiles all the time via `pprof` endpoints?

File: pkg/trace/agent/run.go (hunk 2 above)
Comment (gbbr) [reply pkg/trace/agent/run.go]: Yes.

File: pkg/trace/flags/flags_windows.go [hunk 4]
```diff
@@ -1,31 +0,0 @@
-// Unless explicitly stated otherwise all files in this repository are licensed
-// under the Apache License Version 2.0.
-// This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2016-2019 Datadog, Inc.
-
-// +build windows
```
Comment (palazzem) [inline pkg/trace/flags/flags_windows.go]: This was specific for Windows. How do we handle that?

File: pkg/trace/agent/run.go (hunk 3 above)
Comment (gbbr) [reply pkg/trace/agent/run.go]: These were never used in practice, we just use the HTTP endpoints to extract profiles whenever we want.

File: pkg/trace/flags/flags_windows.go [hunk 5]
```diff
@@ -1,31 +0,0 @@
...
-
-import (
-	"flag"
-	"path/filepath"
-
-	"github.com/DataDog/datadog-agent/pkg/util/winutil"
-)
-
-// DefaultConfigPath specifies the default configuration path.
-var DefaultConfigPath = "c:\\programdata\\datadog\\datadog.yaml"
```
Comment (palazzem) [inline pkg/trace/flags/flags_windows.go]: Not sure how do we pass this as default config. What is the new default config path for Windows?

File: pkg/trace/flags/flags_windows.go (hunk 4 above)
Comment (gbbr) [reply pkg/trace/flags/flags_windows.go]: This was just moved to `cmd/trace-agent` now.

File: pkg/trace/flags/flags_windows.go (hunk 5 above)
Comment (gbbr) [reply pkg/trace/flags/flags_windows.go]: Same here, just search the page, you'll find it.

**Comment by albertvaka** [comment]
Set the milestone to `triage` since this is `do-not-merge/WIP`.

**Comment by gzussa** [comment]
Adding Intg Tools and Library team as reviewers just so that we know when this get merged. We would like to use this feature to simplify some of our scripts for https://github.com/DataDog/datadog-cloudfoundry-buildpack

**Comment by gbbr** [comment]
Closing.

**Pull request closed without merging**